portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -2863,6 +2870,61 @@ private void _inheritRolesPermissions(
 		}
 	}
 
+	private void _initializeFileEntryPermissionForDLFolder(
+			long folderId, DLFileEntry dlFileEntry)
+		throws PortalException {
+
+		int fileEntriesCount = getFileEntriesCount(
+			dlFileEntry.getGroupId(), dlFileEntry.getFolderId());
+
+		if (fileEntriesCount != 1) {
+			return;
+		}
```

Comment:
I don't think we need to check the number of documents in the folder here. Conditions at L2900 are sufficient